maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -1079,6 +1084,33 @@ private String getResumeFrom( List<MavenProject> mavenProjects, MavenProject fai
         return ":" + failedProject.getArtifactId();
     }
 
+    private boolean createResumeFromCacheFile( MavenExecutionResult result, String resumeFromProject )
+    {
+        String buildDirectory = result.getProject().getBuild().getDirectory();
+        File buildDirectoryFile = new File( buildDirectory );
+        boolean isBuildDirectoryPresent = ( buildDirectoryFile.exists() || buildDirectoryFile.mkdirs() );
+        if ( isBuildDirectoryPresent )
+        {
+            Path resumeFromCache = Paths.get( buildDirectory, "resume-from-cache" );
+            try
+            {
+                Files.write( resumeFromCache, resumeFromProject.getBytes() );
```

Comment:
Will work for now, but the MavenExecutionRequest might contain more information worth storing

Reply:
Right, do you think we should make it in a more extensible format like json already? Or should we refactor it when we see the need for other bits of info?

Reply:
Maven doesn't have a json parser. Properties would make more sense here.

Reply:
Ok, but do you think we should implement this already, or only when we need more data there?